internal/adapters/operation/operation_flow_redis_test.go

```diff
@@ -270,3 +274,159 @@ func TestWatchOperationCancellationRequests(t *testing.T) {
+				schedulerName: "test-scheduler",
+			},
+			environmentSetup{
+				prepareDatabase: func(schedulerName string, client *redis.Client) {
+					err := client.RPush(context.Background(), fmt.Sprintf("pending_operations:%s", schedulerName), "some-op-id").Err()
+					require.NoError(t, err)
+				},
+				forceClientError: false,
+			},
+			[]string{"some-op-id"},
+			nil,
+		},
+		{"return no error and the list of pending operations from aux queue",
+			args{
+				ctx:           context.Background(),
+				schedulerName: "test-scheduler",
+			},
+			environmentSetup{
+				prepareDatabase: func(schedulerName string, client *redis.Client) {
+					err := client.RPush(context.Background(), fmt.Sprintf("pending_operations:%s", schedulerName), "some-op-id2").Err()
+					require.NoError(t, err)
+					err = client.RPush(context.Background(), fmt.Sprintf("pending_operations:%s:auxiliary", schedulerName), "some-op-id1").Err()
+					require.NoError(t, err)
+				},
+				forceClientError: false,
+			},
+			[]string{"some-op-id1", "some-op-id2"},
+			nil,
+		},
+		{"return error when some error occurs with redis client",
+			args{
+				ctx:           context.Background(),
+				schedulerName: "test-scheduler",
+			},
+			environmentSetup{
+				prepareDatabase:  func(schedulerName string, client *redis.Client) {},
+				forceClientError: true,
+			},
+			[]string{"some-op-id1", "some-op-id2"},
+			errors.NewErrUnexpected("failed to list pending operations for \"test-scheduler\" from main queue: redis: client is closed"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := test.GetRedisConnection(t, redisAddress)
+			flow := NewRedisOperationFlow(client)
+
+			ctx := tt.args.ctx
+			schedulerName := tt.args.schedulerName
+
+			tt.environmentSetup.prepareDatabase(schedulerName, client)
+
+			if tt.environmentSetup.forceClientError {
+				client.Close()
+				_, err := flow.ListSchedulerPendingOperationIDs(ctx, schedulerName)
+				assert.EqualError(t, err, tt.wantErr.Error())
+			} else {
+				gotOperationsIDs, err := flow.ListSchedulerPendingOperationIDs(ctx, schedulerName)
+				if tt.wantErr != nil {
+					assert.EqualError(t, err, tt.wantErr.Error())
+				}
+				assert.Equal(t, tt.wantOperationsIDs, gotOperationsIDs)
+			}
+		})
+	}
+}
+
+func TestRemoveOperation(t *testing.T) {
+
+	type args struct {
+		ctx           context.Context
+		schedulerName string
+	}
+	type environmentSetup struct {
+		numberOfOps      int
+		forceClientError bool
+	}
+	tests := []struct {
+		name             string
+		args             args
+		environmentSetup environmentSetup
+	}{
+		{"return no error and pops the operation from de auxiliary operations queue if there is some", args{
+			ctx:           context.Background(),
+			schedulerName: "test-scheduler",
+		},
+			environmentSetup{numberOfOps: 1, forceClientError: false},
+		},
+		{"return no error and does nothing if the auxiliary operations queue is empty", args{
```

Comment:
Hey in this case we should not send an error saying that the user is trying to remove an inexistent operation? It seems like an error.